### PR TITLE
Upgrade socket2 to 0.5

### PR DESCRIPTION
**Pull request by djc**
This should be okay now that we've bumped our MSRV?

### CODE REVIEW

**Comment by djc** [comment]
The CI failure here is confusing. Tokio's support for `AsFd` was added in https://github.com/quinn-rs/quinn/actions/runs/4924706904/jobs/8798057112?pr=1558 (released in 1.27), and we already have 1.63 as our MSRV.

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -77,7 +77,7 @@ jobs:
       - uses: actions-rs/cargo@v1
         with:
           command: check
-          args: --lib --all-features -p quinn-udp -p quinn-proto -p quinn
```
Comment (Ralith) [inline .github/workflows/rust.yml]: Stray commit?

**Review by Ralith — CHANGES_REQUESTED** [review]
Great improvements overall!

**Comment by djc** [comment]
Bumped the tokio dependency in quinn to 1.28.1 which should make the MSRV test pass.

**Review by Ralith — APPROVED** [review]

**Pull request merged**